src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -882,6 +885,20 @@ def load_module(name, value):
                     f"Expected types for {key}: {_expected_class_types}, got {class_obj.__class__.__name__}."
                 )
 
+        # Check `torch_dtype` map for unused keys
+        if isinstance(torch_dtype, dict):
+            extra_keys_dtype = set(torch_dtype.keys()) - set(passed_class_obj.keys())
+            extra_keys_obj = set(passed_class_obj.keys()) - set(torch_dtype.keys())
+            if len(extra_keys_dtype) > 0:
+                logger.warning(
+                    f"Expected `{list(passed_class_obj.keys())}`, got extra `torch_dtype` keys `{extra_keys_dtype}`."
+                )
+            if len(extra_keys_obj) > 0:
+                logger.warning(
```

Comment:
I don't think we need this warning. I think the expectation of passed class objects is that their dtype is already set and if it isn't it happens at the model level where a `dtype=None` results in FP32 default.